scrapy/pqueues.py

```diff
@@ -157,11 +161,15 @@ def __init__(self, crawler, downstream_queue_cls, key, slot_startprios=()):
         self.downstream_queue_cls = downstream_queue_cls
         self.key = key
         self.crawler = crawler
-
+        self.selfcheck()
         self.pqueues = {}  # slot -> priority queue
         for slot, startprios in (slot_startprios or {}).items():
             self.pqueues[slot] = self.pqfactory(slot, startprios)
 
+    def selfcheck(self):
+        queue = self.pqfactory('hostname.invalid')
+        queue.close()
```

Comment:
:lipstick: In line with the equivalent from the previous queue:

```suggestion
        self.pqfactory('hostname.invalid').close()
```

Reply:
Fixed, thanks!